src/Storages/StorageS3.cpp

```diff
@@ -28,6 +28,8 @@
 #include <Storages/getVirtualsForStorage.h>
 #include <Storages/checkAndGetLiteralArgument.h>
 #include <Storages/StorageURL.h>
+#include <Storages/NamedCollections/NamedCollectionsHelpers.h>
```

Comment:
Why NamedCollections are in the /Storages folder? I expect them to be generic and be dependent only on AST

Reply:
just because they are mostly used for storages, can be moved easily